.github/workflows/ci.yml

```diff
@@ -62,7 +62,12 @@ jobs:
           # and the ^ prefix is used to exclude these parents and all their
           # ancestors from the rev-list output as described in:
           # https://git-scm.com/docs/git-rev-list
-          echo "TEST_BASE=$(git rev-list -n$((${{ env.MAX_COUNT }} + 1)) --reverse HEAD ^$(git rev-list -n1 --merges HEAD)^@ | head -1)" >> "$GITHUB_ENV"
+          MERGE_BASE=$(git rev-list -n1 --merges HEAD)
+          EXCLUDE_MERGE_BASE_ANCESTORS=
+          if test -n "$MERGE_BASE"; then
```

Comment:
In commit "ci: test-each-commit merge base optional" (65abc0b5b54edb06cc3bf584691beddbeb802ddc)

Maybe add comment like "MERGE_BASE can be empty due to limited fetch-depth". Otherwise I don't think it's clear why git wouldn't be able to find the merge commit.